chapter_optimization/adagrad.md

```diff
@@ -139,13 +147,28 @@ def adagrad(params, states, hyperparams):
         p.grad.data.zero_()
 ```
 
+```{.python .input}
+#@tab tensorflow
+def init_adagrad_states(feature_dim):
+    s_w = tf.Variable(d2l.zeros((feature_dim, 1)))
+    s_b = tf.Variable(d2l.zeros(1))
+    return (s_w, s_b)
+
+def adagrad(params, grads, states, hyperparams):
+    eps = 1e-6
+    for p, s, g in zip(params, states, grads):
```

Comment:
There is a bug over here, please help if you guys can figure it. @terrytangyuan @astonzhang

Reply:
Fixed.